content/pages/nuttx/getting-started/index.md

```diff
@@ -0,0 +1,279 @@
+```
+
+Also, please install some python packages:
+
+```bash
+pip install kconfiglib
+```
+
+#### Espressif Tools and Toolchains
+
+Once the NuttX dependencies are available, it’s time to install the compilation tools for Espressif devices.
+
+Install `esptool`:
+
+```bash
+pip install esptool==4.8.dev4
+```
+Finally, install the toolchains to build the Espressif SoCs. It can be installed depending on the SoC being used, but it is highly recommended to keep all of them installed.
+
+##### **Xtensa Devices (ESP32, ESP32-S2, ESP32-S3)**
+
+Each Xtensa-based device has its own toolchain, which needs to be downloaded and configured separately. All the toolchains used by NuttX can be seen in the Docker container used for upstream testing. It is highly recommended to check the updated versions according to the file in https://github.com/apache/nuttx/tree/master/tools/ci/docker/linux/Dockerfile
+
+###### **ESP32**:
+
+Create a directory to install the toolchain. It can be on your home directory:
+```bash
+mkdir -p ~/xtensa-esp32-elf-gcc
+```
+
+Download and extract the toolchain on the new directory:
+```bash
+cd ~/xtensa-esp32-elf-gcc
+wget https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/xtensa-esp32-elf-12.2.0_20230208-x86_64-linux-gnu.tar.xz
+tar -xf xtensa-esp32-elf-12.2.0_20230208-x86_64-linux-gnu.tar.xz
+```
+
+Once extracted, add the location of the binaries to the user’s PATH and reload the terminal:
+
+```bash
+echo "export PATH=$HOME/xtensa-esp32-elf-gcc/xtensa-esp32-elf/bin:\$PATH" >> ~/.bashrc
+```
+
+###### **ESP32-S2**:
+
+Create a directory to install the toolchain. It can be on your home directory:
+```bash
+mkdir -p ~/xtensa-esp32s2-elf-gcc
+```
+
+Download and extract the toolchain on the new directory:
+```bash
+cd ~/xtensa-esp32s2-elf-gcc
+wget https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/xtensa-esp32s2-elf-12.2.0_20230208-x86_64-linux-gnu.tar.xz
+tar -xf xtensa-esp32s2-elf-12.2.0_20230208-x86_64-linux-gnu.tar.xz
+```
+
+Once extracted, add the location of the binaries to the user’s PATH and reload the terminal:
+```bash
+echo "export PATH=$HOME/xtensa-esp32s2-elf-gcc/xtensa-esp32s2-elf/bin:\$PATH" >> ~/.bashrc
+```
+
+###### **ESP32-S3**:
+
+Create a directory to install the toolchain. It can be on your home directory:
+```bash
+mkdir -p ~/xtensa-esp32s3-elf-gcc
+```
+
+Download and extract the toolchain on the new directory:
+```bash
+cd ~/xtensa-esp32s3-elf-gcc
+wget https://github.com/espressif/crosstool-NG/releases/download/esp-12.2.0_20230208/xtensa-esp32s3-elf-12.2.0_20230208-x86_64-linux-gnu.tar.xz
+tar -xf xtensa-esp32s3-elf-12.2.0_20230208-x86_64-linux-gnu.tar.xz
+```
+
+Once extracted, add the location of the binaries to the user’s PATH and reload the terminal:
+
+```bash
+echo "export PATH=$HOME/xtensa-esp32s3-elf-gcc/xtensa-esp32s3-elf/bin:\$PATH" >> ~/.bashrc
+```
+
+##### **RISC-V Devices (ESP32-C3, ESP32-C6, ESP32-H2)**
```

Comment:
I would consider moving RISC-V targets first, before Xtensa, because they're included in all new chips and should be kind of default option.

Reply:
done!